Gordon360/Controllers/HousingController.cs

```diff
@@ -229,4 +235,463 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
+    [HttpGet]
+    [Route("roomrange/assignment/all")]
+    [StateYourBusiness(operation = Operation.READ_ALL, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<IActionResult> GetRangeAssignments()
+    {
+        try
+        {
+            var RangeAssignments = await housingService.GetRangeAssignmentsAsync();
+            if (RangeAssignments == null)
+            {
+                return NotFound("No Assigned Ranges");
+            }
+            return Ok(RangeAssignments);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Deletes an RA range assignment
+    /// </summary>
+    /// <param name="rangeId">The Room range of the assignment to delete</param>
+    /// <returns> Returns if completed</returns>
+    [HttpDelete("roomrange/assignment/{rangeId}")]
+    [StateYourBusiness(operation = Operation.DELETE, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<IActionResult> DeleteAssignment(int rangeId)
+    {
+        try
+        {
+            var success = await housingService.DeleteAssignmentAsync(rangeId);
+            if (success)
+            {
+                return Ok(new { message = "Assigment deleted successfully." });
+            }
+            else
+            {
+                return BadRequest(new { message = "Failed to delete the Assignment." });
+            }
+        }
+        catch (ResourceNotFoundException ex)
+        {
+            return NotFound(new { message = ex.ExceptionMessage });
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieve the RD of the resident's hall based on their hall ID.
+    /// </summary>
+    /// <param name="hallId">The ID of the hall.</param>
+    /// <returns>Returns the RD's details if found, otherwise null.</returns>
+    [HttpGet("rd/{hallId}")]
+    public async Task<IActionResult> GetResidentRD([FromRoute] string hallId)
+    {
+        try
+        {
+            var rdInfo = await housingService.GetResidentRDAsync(hallId);
+            return Ok(rdInfo);
+        }
+        catch (InvalidOperationException ex)
+        {
+            return NotFound(ex.Message); // Return 404 if no RD is found for the specified hall
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the RA assigned to a resident based on their room number and hall ID.
+    /// </summary>
+    /// <param name="hallId">The ID of the hall.</param>
+    /// <param name="roomNumber">The resident's room number.</param>
+    /// <returns>Returns the RA's details if found, otherwise null. </returns>
+    [HttpGet("ra/{hallId}/{roomNumber}")]
```

Comment:
The idiomatic route would be `/RAs?hallId={hallId}&roomNumber={roomNumber}`.

Reply:
added to new branch